common/containers/Tabs/SendTransaction/components/ConfirmationModal.jsx

```diff
@@ -56,24 +72,51 @@ export default class ConfirmationModal extends React.Component {
       }
     }, 1000);
 
-    this.props.wallet.getAddress().then(address => {
-      this.setState({ address });
-    });
+    this._setWalletAddress(this.props.wallet);
   }
 
   componentWillUnmount() {
     clearTimeout(this.readTimer);
   }
 
+  _getStateFromProps(props: Props) {
+    const { rawTransaction, token } = props;
+    const { value, gasPrice, nonce, to } = rawTransaction;
+    let fixedValue;
+    let toAddress;
+
+    if (token) {
```

Comment:
Can you elaborate? Are you suggesting I figure this out from to address / transaction data?